packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/simulationservice/SimulationRequestResource.java

```diff
@@ -177,5 +192,42 @@ public JobResponse makeEnsembleCalibrateCiemssJob(
 		return res;
 	}
 
-
+	//Get modelConfigId
+	//Check if it has timeseries in its metadata
+	//If it does for each element convert it to type Intervention and add it to this.interventions
+	//Schema: http://json-schema.org/draft-07/schema#
+	public List<Intervention> getInterventionFromId(String modelConfigId){
+		List<Intervention> interventionList = new ArrayList<>();
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			ModelConfiguration modelConfig = modelConfigProxy.getModelConfiguration(modelConfigId);
+			JsonNode configuration =  mapper.convertValue(modelConfig.getConfiguration(), JsonNode.class);
+			if (configuration.findValue("timeseries") != null){
```

Comment:
Can we be more specific here? There are a number of things that can also be called "timeseries" in the overall structure of the model, since "groundings" and "modifiers" field are kind of freeform it might be the case that they can contain "timeseries" as well.

Reply:
Good call I can go through the full path to get to the correct key